cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -170,16 +170,18 @@ public void apply() throws Exception {
     }
   }
 
-  private void executePersist(final TriggerKey triggerKey, final Trigger.TriggerState state) {
+  private void executePersist(final TriggerKey triggerKey, final Trigger.TriggerState newTriggerState) {
     try {
       factory.createExecutor(ImmutableList.of((TransactionAware) table))
                             .execute(new TransactionExecutor.Subroutine() {
                               @Override
                               public void apply() throws Exception {
                                 if (triggerKey != null) {
-                                  TriggerStatus trigger = readTrigger(triggerKey);
-                                  if (trigger != null) {
-                                    persistTrigger(table, trigger.trigger, state);
+                                  TriggerStatus storedTriggerStatus = readTrigger(triggerKey);
+                                  if (storedTriggerStatus != null) {
```

Comment:
This executePersist for triggers is only called from pause and resume. So the TriggerStatus(our wrapper on trigger with its status) has to be present while we perform pause or resume to it. It will be persisted initially when the trigger is added.